### PR TITLE
feat(coding-guidelines): add guideline for C++

### PR DESCRIPTION
**Pull request by isamu-takagi**
Signed-off-by: Takagi, Isamu <isamu.takagi@tier4.jp>

## Description

Add guideline for C++.
Related to https://github.com/autowarefoundation/autoware/discussions/2824.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The Reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by kenji-miyake** [comment]
@isamu-takagi May I update this PR to be a similar style to AUTOSAR's guidelines and C++ Core Guidelines?

**Comment by isamu-takagi** [comment]
@kenji-miyake Sounds good. Can you please?

**Comment by kenji-miyake** [comment]
@isamu-takagi Pushed my proposals. How about this?

**Comment by isamu-takagi** [comment]
@kenji-miyake Nice work! I think it's good.

**Comment by kenji-miyake** [comment]
Okay, then @xmfcx Could you briefly take a look?

**Review by kenji-miyake — APPROVED** [review]

**Review by xmfcx — APPROVED** [review]

File: docs/contributing/coding-guidelines/languages/cpp.md [hunk 1]
```diff
@@ -4,8 +4,195 @@
...
+#### Reference
+
+- <https://en.cppreference.com/w/cpp/numeric/constants>
+
+#### Example
+
+```cpp
+constexpr double gravity = 9.80665;
+```
+
```
Comment (xmfcx) [inline docs/contributing/coding-guidelines/languages/cpp.md:182]: ROS2 seems to be using this convention as well [(example)](https://github.com/ros2/rclcpp/blob/33dae5d679751b603205008fcb31755986bcee1c/rclcpp/src/rclcpp/parameter_service_names.hpp#L23), I'm ok with it too.

**Review by xmfcx — APPROVED** [review]

File: docs/contributing/coding-guidelines/languages/cpp.md (hunk 1 above)
Comment (kenji-miyake) [reply docs/contributing/coding-guidelines/languages/cpp.md]: Thanks! Updated in 2cb800e1db8f3ce87a46d669594f9925fa1d68c2.

**Pull request merged**